plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/machine/agent/launcher/ExternalRsyncAgentLauncherImpl.java

```diff
@@ -41,7 +43,10 @@
     @Inject
     public ExternalRsyncAgentLauncherImpl(@Named("che.agent.dev.max_start_time_ms") long agentMaxStartTimeMs,
                                           @Named("che.agent.dev.ping_delay_ms") long agentPingDelayMs) {
-        super(agentMaxStartTimeMs, agentPingDelayMs, AgentLaunchingChecker.DEFAULT);
+        super(agentMaxStartTimeMs,
+              agentPingDelayMs,
+              new CompositeAgentLaunchingChecker(new CommandExistsAgentChecker("rsync"),
+                                                 AgentLaunchingChecker.DEFAULT));
```

Comment:
are you sure it's good idea to create objects here, may be we should delegate it to container?